opds_import.py

```diff
@@ -288,53 +360,148 @@ def import_from_metadata(
+
+        policy = ReplacementPolicy(
+            subjects=True,
+            links=True,
+            contributions=True,
+            rights=True,
+            link_content=True,
+            even_if_not_apparently_updated=True,
+            mirror=self.mirror,
+            http_get=self.http_get,
+        )
+        circulationdata.apply(
+            license_pool, replace=policy
+        )
+
         if license_pool is None:
             # Without a LicensePool, we can't create a Work.
-            self.log.warn(
-                "No LicensePool present for Edition %r, not attempting to create Work.",
-                edition
-            )
-        else:
-            license_pool.presentation_edition = edition
-            work, is_new_work = license_pool.calculate_work(
-                known_edition=edition,
-                even_if_no_author=even_if_no_author,
-            )
-            if work:
-                if immediately_presentation_ready:
-                    # We want this book to be presentation-ready
-                    # immediately upon import. As long as no crucial
-                    # information is missing (like language or title),
-                    # this will do it.
-                    work.set_presentation_ready_based_on_content()
+            error_message = "No LicensePool created for CirculationData %r, will not attempt to continue to create Work." % circulationdata
+            self.log.warn(error_message)
+            raise DBImportException(error_message)
+
+        return license_pool
 
-        return edition
 
-    def extract_metadata(self, feed):
-        """Turn an OPDS feed into a list of Metadata objects and a list of
-        messages.
+    def import_work_from_circulationdata(
+            self, license_pool, even_if_no_author, immediately_presentation_ready
+    ):
+        """ Making work is not based on pool's cutoff logic, but based on whether have a pool and and edition, period.
+        Don't mind if they're new or not.
+        """
+
+        # Note: pool.calculate_work will call self.set_presentation_edition(), 
+        # which will find editions attached to same Identifier.
+        # If the pool has no edition, pool.calculate_work is responsible for not allowing the work to be created.
+        work = None
+        work, is_new_work = license_pool.calculate_work(even_if_no_author=even_if_no_author,)
+        # if pool.calculate_work made a new work, it already called calculate_presentation
+        if (work and not is_new_work):
+            work.calculate_presentation()
+        if (work):
+            if immediately_presentation_ready:
+                # We want this book to be presentation-ready
+                # immediately upon import. As long as no crucial
+                # information is missing (like language or title),
+                # this will do it.
+                work.set_presentation_ready_based_on_content()
+        return license_pool, work
```

Comment:
it might have.  the pool gets updated with the new work.  having said that, the pool is passed in, so doesn't need to be returned.  having said _that_, I like returning it, to make it a bit more clear that stuff possibly happened to the pool.